src/components/CustomInputs/DictInput.vue

```diff
@@ -57,11 +63,23 @@ export default {
       return this.keys.map((p, i) => i)
     },
     value() {
-      const dict = {}
-      this.keys
-        .filter(k => k !== null)
-        .map((k, i) => (dict[k] = this.values[i]))
-      return dict
+      const parametersDict = []
+      const obj = Object.assign(
+        {},
+        ...this.keys
+          .filter(k => k !== null)
+          .map((e, i) => ({ [e]: this.values[i] }))
+      )
+      for (const [key, value] of Object.entries(obj)) {
+        if (this.includeCheckbox) {
+          if (this.paramsToAdd.includes(key)) {
+            parametersDict.push({ [key]: value })
+          }
+        } else {
+          parametersDict.push({ [key]: value })
+        }
+      }
+      return Object.assign({}, ...parametersDict)
```

Comment:
This isn't the correct place for this or at least is incorrectly and confusingly named for what the component is meant to do; this block, and the `paramsToAdd` prop, are specific to the parameters implementation only and mean that an otherwise generic component/interface has to know something about its parent. This is also a really confusing block, couldn't we write it like this?

```suggestion
      const dict = {}
      this.keys
        .filter(k => k !== null)
        .filter(k => !this.includeCheckbox || this.paramsToAdd.includes(k))
        .map((k, i) => (dict[k] = this.values[i]))
      return dict
```

Reply:
For the suggested code,  if you have two or more parameters and then select only one parameter then the index will be at the 0 position and it will give the wrong value which is what the code above was trying to change. I agree that I think it could be written better

Reply:
We also included a `includeCheckbox`prop which I think that would give the component the ability to opt out?

Reply:
Oops good point, you could do this instead:
```
      const dict = {}
      this.keys.forEach((k, i) => {
        if (k && (!this.includeCheckbox || this.paramsToAdd.includes(k))) {
          dict[k] = this.values[i]
        }
      })
      return dict
```

Reply:
To your second point, yes, the prop will allow an instantiation to ignore that logic but the point is that the logic should be generic anyway, i.e. shouldn't contain references to a specific data instance. The component should be able to be read as completely agnostic to the business logic but using variables and methods that are implementation-specific (even if they're opt-out) hinders that readability. Params !== keys and mean the component can't be taken in isolation without understanding what parameters are. In this case the swap is pretty easy, since changing the variable/method names is probably all that's required. So instead of `paramsToInclude`, something like `includedKeys` is agnostic to any external data structure.

Reply:
ahhh okay that makes sense